File: Programmers/Greedy/체육복.py
def solution(n, lost, reserve):
    answer = 0
    newlost, newreserve = [i for i in lost if i not in reserve], [i for i in reserve if i not in lost]
    for i in range(len(newlost)):
        for j in range(len(newreserve)):
            if newlost[i] - 1 == newreserve[j] or newlost[i] + 1 == newreserve[j]:
                newreserve[j] = -2
                answer+=1
                break
    return n-len(newlost)+answer

File: Programmers/Greedy/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    for(int i = 0; i<lost.size(); i++)
        for(int j = 0; j<reserve.size();j++)
            if(lost[i] == reserve[j]){
                lost[i] = -1;
                reserve[j] = -1;
                break;
            }
    for(int i = 0; i<lost.size(); i++)
        for(int j = 0; j<reserve.size();j++)
            if((lost[i] != -1 && reserve[j] != -1) && (lost[i]-1 == reserve[j] || lost[i]+1 == reserve[j]))
            {
                lost[i] = -1;
                reserve[j] = -1;
                break;
            }
    
    for(int i = 0; i < lost.size(); i++)
        if(lost[i] != -1)
            n -= 1;        
    
    return n;
}